paperqa/agents/task.py

```diff
@@ -487,3 +493,229 @@ def __len__(self) -> int:
     LitQAv2TaskDataset.__module__,
     LitQAv2TaskDataset.__name__,
 )
+
+
+class LFRQAPairwiseEvalEnv(GradablePaperQAEnvironment):
+    def __init__(
+        self,
+        *args,
+        qid: str | UUID,
+        question: str,
+        human_answer: str,
+        gt_doc_ids: list[str],
+        pairwise_eval_llm: LLMModel | str = CommonLLMNames.GPT_4O.value,
+        **kwargs,
+    ):
+        # NOTE I'm using qid and question and
+        # maybe we could use session_id and query instead?
+        kwargs["query"] = question
+
+        # NOTE: I'm also passing this docs because they are required
+        # but don't know if/how they are used
+        kwargs["docs"] = Docs()
+        super().__init__(*args, **kwargs)
+
+        self.qid = qid
+        self.question = question
+        self.human_answer = human_answer
+        self.gt_doc_ids = gt_doc_ids
+        self.pairwise_eval_llm = pairwise_eval_llm
+
+    def extract_best_answer_index(self, text: str) -> int:
+        match = re.search(r"<rating>(\d+)</rating>", text)
+        return int(match.group(1)) if match else 0
+
+    def log_results_to_json(
+        self,
+        llm_model_name: str,
+        qid: str | UUID,
+        question: str,
+        pqa_answer: str,
+        paper_search_ids: list[int],
+        pqa_context: str,
+        human_answer: str,
+        gt_doc_ids: list[str],
+        pqa_answer_index: int,
+        winner: str,
```

Comment:
Can we use a `StrEnum` or `Enum` over `str`, since there's a set number of players